lib/CrashHandler/CrashHandler.cpp

```diff
@@ -179,28 +185,37 @@ void appendHexValue(unsigned char const* src, size_t len, char*& dst,
 }
 
 template<typename T>
-void appendHexValue(T value, char*& dst, bool stripLeadingZeros) {
+void appendHexValue(T value, size_t bytesLeft, char*& dst,
+                    bool stripLeadingZeros) {
+  if (sizeof(T) * 2 < bytesLeft) {
+    return;
+  }
   static_assert(std::is_integral_v<T> || std::is_pointer_v<T>);
   unsigned char buffer[sizeof(T)];
   memcpy(buffer, &value, sizeof(T));
   appendHexValue(buffer, sizeof(T), dst, stripLeadingZeros);
 }
 
 #ifdef ARANGODB_HAVE_LIBUNWIND
-void appendAddress(unw_word_t pc, long base, char*& dst) {
+void appendAddress(unw_word_t pc, long base, size_t bytesLeft, char*& dst) {
+  char* begin = dst;
   if (base == 0) {
     // absolute address of pc
-    appendNullTerminatedString(" [$0x", dst);
-    appendHexValue(reinterpret_cast<unsigned char const*>(&pc),
-                   sizeof(decltype(pc)), dst, false);
+    appendNullTerminatedString(" [$0x", bytesLeft, dst);
+    if (bytesLeft - (dst - begin) >= 2 * sizeof(decltype(pc))) {
```

Comment:
The left side of the comparison can underflow if `bytesLeft < (dst - begin)`, and then it will always evaluate to `true`.

Reply:
It should rather be `if (bytesLeft >= (dst - begin) + 2 * sizeof(decltype(pc)))`

Reply:
Since we used `bytesLeft` in the first call to `appendNullTerminatedString`, `(dst - begin)` can never be larger than `bytesLeft`, therefore the difference cannot underflow.

Reply:
Fixed to make it more understandable.